jinahub/encoders/text/TransformerTorchEncoder/transform_encoder.py

```diff
@@ -62,22 +62,33 @@ def __init__(
         if not device in ['cpu', 'cuda']:
             self.logger.error('Torch device not supported. Must be cpu or cuda!')
             raise RuntimeError('Torch device not supported. Must be cpu or cuda!')
+
         if device == 'cuda' and not torch.cuda.is_available():
             self.logger.warning(
-                'You tried to use GPU but torch did not detect your'
+                'You tried to use GPU but torch did not detect your '
                 'GPU correctly. Defaulting to CPU. Check your CUDA installation!'
             )
             device = 'cpu'
 
+        if device == 'cuda':
```

Comment:
I am not sure this is the way to go. If we were serious about this, we need to add this to every single pytorch executor. And then we need to think of something similar for tensorflow executors, paddle paddle executors, etc. Another issue that I have with this is that someone might want to have multiple smaller models on a single GPU - seems like this right now is doable by specifying `cuda:0` - but such strong difference between `cuda` and `cuda:0` is not what pytorch users would expect.

To me it feels like this is something that should be handled on a lower (or higher, as you prefer) level - in jina core. Something along the lines of a possibility of passing different init arguments to each replica? I would like to hear someone elses' opinion on this

Reply:
@tadejsv I understand what's your point. I think this PR is a temporal one as a hotfix.  Because I need to fix the multi-gpu issue in one of our EAP projects. I think we should take more time to work on this issue. **A better generalized solution should be the future. **

Reply:
@numb3r3 If it is nececssary, I am not completely against merging it in as a hotfix. But what about the alternative of, for now, copying this code and creating a custom executor in the EAP project? If that is not too feasible we can merge it

Reply:
> copying this code and creating a custom executor in the EAP project?

Yes, this can solve my problem. But I still want to encourage our EAP partners to use `jinahub`.

Reply:
Ok then, let's do this temporary hotfix then. But we should ping @JoanFM once he gets back from vacation to take a look at this, as we would need a permanent solution also